internal/cli/cmd/install.go

```diff
@@ -91,6 +105,20 @@ cilium install --context kind-cluster1 --cluster-id 1 --cluster-name cluster1
 	cmd.Flags().StringArrayVar(&params.HelmOpts.FileValues, "helm-set-file", []string{}, "Set helm values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)")
 	cmd.Flags().StringVar(&params.HelmGenValuesFile, "helm-auto-gen-values", "", "Write an auto-generated helm values into this file")
 
+	for flagName, helmOpt := range install.DeprecatedFlagsToHelmOpts {
+		msg := fmt.Sprintf("use --helm-set=%s<=value> instead", helmOpt)
+		err := cmd.Flags().MarkDeprecated(flagName, msg)
+		if err != nil {
+			panic(err)
+		}
+		install.DeprecatedFlagsValues[flagName] = cmd.Flags().Lookup(flagName).Value
+	}
+	err := cmd.Flags().MarkDeprecated("config", "use --helm-set=extraConfig.<key>=<value>[,extraConfig.<key>=<value>] instead")
```

Comment:
Personally I think `--config` is a useful short-hand to set a value in the `cilium-config` CM from the command line. I'd thus be inclined to keep this as an alias and not deprecate it.

Also, as mentioned in https://github.com/cilium/cilium-cli/pull/717#issuecomment-1068830584 we also use `--config` in our getting started guides. So users would get deprecation warnings when using commands copied from the guides. I case we decide to deprecate `--config` I think we should at least make sure to update the guides for v1.10/v1.11/master to use the new flag syntax.